src/java/org/apache/cassandra/index/sai/disk/vector/CompactionGraph.java

```diff
@@ -256,12 +251,11 @@ public InsertionResult maybeAddVector(ByteBuffer term, int segmentRowId) throws
             postings = new CompactionVectorPostings(ordinal, segmentRowId);
             postingsMap.put(vector, postings);
             writer.writeInline(ordinal, Feature.singleState(FeatureId.INLINE_VECTORS, new InlineVectors.State(vector)));
-            var encoded = (ArrayByteSequence) compressor.encode(vector);
-            while (pqVectorsList.size() < ordinal)
-                pqVectorsList.add(encodedOmittedVector);
-            pqVectorsList.add(encoded);
+            // Fill in any holes in the pqVectors (setZero has the side effect of increasing the count)
+            while (pqVectors.count() < ordinal)
+                pqVectors.setZero(pqVectors.count());
+            pqVectors.encodeAndSet(ordinal, vector);
 
-            bytesUsed += encoded.ramBytesUsed();
```

Comment:
The bytes are allocated when we create the PQVectors class. Therefore, encoding and setting the pq vector does not increase memory utilization.